app/services/foreman_ansible/ansible_info.rb

```diff
@@ -14,5 +18,19 @@ def ansible_params
         memo
       end
     end
+
+    def ansible_roles
+      roles = host.all_ansible_roles.pluck(:id, :name)
+
+      roles.each_with_object({}) do |role, rmemo|
+        variables = AnsibleVariable.where(:ansible_role_id => role[0], :override => true)
```

Comment:
The content of the whole block looks very similar to what is in `ansible_params` method above. Do you think it would be possible to extract it into a separate method used by both?

Reply:
You're right. I'll take look and see if I can do that.

Reply:
Based on what @ares said about only having one behavior, I pulled out the old method.